atlascope/core/rest/endpoints/dataset_endpoints.py

```diff
@@ -44,3 +50,21 @@ def create(self, request):
         )
 
         return Response(DatasetSerializer(new_dataset_obj).data, status=status.HTTP_201_CREATED)
+
+    @swagger_auto_schema(request_body=DatasetSubImageSerializer())
+    @action(detail=False, methods=['POST'])
+    def createSubImage(self, request):
+        serializer = DatasetSubImageSerializer(data=self.request.data)
+        serializer.is_valid(raise_exception=True)
+        original_dataset = Dataset.objects.get(id=serializer.validated_data["original_dataset_id"])
+        meta = serializer.validated_data
+        meta.pop("original_dataset_id")
```

Comment:
This part confuses the heck out of me. It appears here that removing the `original_dataset_id` from meta also affects the `serializer.validated_data`. Can someone help me understand what is happening here? @waxlamp @annehaley

Reply:
My first impression is that `serializer.validated_data` is a mutable type, so `meta = serializer.validated_data` simply creates an alias. When you `meta.pop()`, you're mutating `meta` which also mutates `serializer.validated_data` (since those are just two names for the same underlying object).

Depending on what you need to do here, you could figure out how to make a copy of the data represented by `validated_data`, and then modify that.

Reply:
Thanks! You were right. I think I have it correct now.

Reply:
A warning: `copy()` does a shallow copy, and that may be sufficient for this case (if `validated_data` only has POD ("plain old datatype") values in it), but in general you would want `deepcopy()`. (See the docs on the [`copy`](https://docs.python.org/3/library/copy.html) module.)